File: backend/src/util/aes.ts
import crypto from 'crypto';

function encrypt(data: string, key: string, iv: string): string {
  if (key.length !== 16 || iv.length !== 16) {
    throw new Error("A String Chave deve ter 16 dígitos");
  }

  const cipher = crypto.createCipheriv('aes-128-cbc', Buffer.from(key, 'utf8'), Buffer.from(iv, 'utf8'));
  let encrypted = cipher.update(data, 'utf8', 'hex');
  encrypted += cipher.final('hex');
  
  return encrypted;
}

function decrypt(data: string, key: string, iv: string): string {
    if (key.length !== 16 || iv.length !== 16) {
      throw new Error("A String Chave deve ter 16 dígitos");
    }
  
    const decipher = crypto.createDecipheriv('aes-128-cbc', Buffer.from(key, 'utf8'), Buffer.from(iv, 'utf8'));
    let decrypted = decipher.update(data, 'hex', 'utf8');
    decrypted += decipher.final('utf8');
    
    return decrypted;
  }

export default {
    encrypt,
    decrypt
}

File: frontend/lib/util/aes.dart
import 'package:encrypt/encrypt.dart' as encrypt;

String aesEncrypt(String data, String key, String iv) {
  final cipherKey = encrypt.Key.fromUtf8(key);
  final cipherIv = encrypt.IV.fromUtf8(iv);

  if (key.length != 16 || iv.length != 16) {
    Exception("A String Chave deve ter 16 dígitos");
  }

  final encrypter = encrypt.Encrypter(
    encrypt.AES(
      cipherKey,
      mode: encrypt.AESMode.cbc,
      padding: 'PKCS7',
    ),
  );

  final encrypted = encrypter.encrypt(data, iv: cipherIv);
  return encrypted.base16;
}

String aesDecrypt(String data, String key, String iv) {
  final cipherKey = encrypt.Key.fromUtf8(key);
  final cipherIv = encrypt.IV.fromUtf8(iv);

  if (key.length != 16 || iv.length != 16) {
    Exception("A String Chave deve ter 16 dígitos");
  }

  final decrypter = encrypt.Encrypter(
    encrypt.AES(
      cipherKey,
      mode: encrypt.AESMode.cbc,
      padding: 'PKCS7',
    ),
  );
  final decrypted = decrypter.decrypt16(data, iv: cipherIv);
  return decrypted;
}
